src/NodeVisitor/OptimizerNodeVisitor.php

```diff
@@ -81,7 +82,9 @@ public function leaveNode(Node $node, Environment $env): ?Node
             $node = $this->optimizeRawFilter($node);
         }
 
-        $node = $this->optimizePrintNode($node);
+        if (self::OPTIMIZE_PRINT === (self::OPTIMIZE_PRINT & $this->optimizers)) {
```

Comment:
Why would we want to be able to disable this optimization. I didn't provide a way to disable it on purpose.

Reply:
I ran into problems with one of my own extension because of missing `SourceContexts` on nodes created by the PrintNode and TextNode optimizations. The obvious solution to me would be to turn off those or all optimizers, but setting the optimizers property on `TwigEnvironment` to 0/`OptimizerNodeVisitor::OPTIMIZE_NONE` still leaves the PrintNode optimization active.

Reply:
As we are fixing the source context, would you agree that there is not need to make this optimization configurable?

Reply:
The source context fix is sufficient to solve my current problems. I still think this should be fixed though as the optimizers option currently doesn't behave in the way it is described in the doc comment in https://github.com/twigphp/Twig/blob/4a7de4ab5603ad8f199222cae3951fc271a8c45d/src/Environment.php#L102 as setting it to `0` does not disable all optimizations.

Reply:
Let's close then.